src/main/java/sg/dex/starfish/Asset.java

```diff
@@ -114,4 +118,22 @@ public default boolean isOperation(){
 	public default boolean isBundle(){
 		return false;
 	}
+
+
+	/**
+	 * This method is to validate the hash of the asset content .
+	 * it will calculate the hash of the content of an Asset
+	 * then compare with the hash value included in metadata,
+	 * if both are not the same , StarfishValidation Exception will be thrown
+	 *
+	 * @throws StarfishValidationException if hash content is not matched , exception will be thrown
+	 */
+	public default void validateContentHash() {
```

Comment:
Looks sensible. Might consider a boolean return value here as an alternative, but I think throwing an exception is fine.